### ISSUE TITLE
Heroku i Postgres

### ISSUE THREAD

**Issue by jmcano94** [UB-GEI-SD/B#16]
Tinc un problema després d'haver fet el deployment a heroku.

Puc obrir la pagina principal de la meva app pero quan accedeixo a qualsevol altre que requereixi de la base de dades, no em deixa i em dona aquest error: 

"could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 5432?"

El mateix em passa quan intento fer "heroku run manage.py migrate". Crec que es problema de la configuració de la base de dades en settings.py. He seguit fil per randa el tutorial que s'ens donava:

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql_psycopg2',
        'NAME': 'sd2018-ykea',
        'USER': 'user',
        'PASSWORD': '',
        'HOST': 'localhost',
        'PORT': '',
        'CONN_MAX_AGE': 500

    }
}

Alguna idea de que esta fallant? Grácies

**Comment by llevaNEUS** [comment]
A veure, una cosa que no queda especificada gaire en el manual es que has d'instal·lar postgres i enllaçar una base de dades al teu projecte. Podria ser això?
Ves al menú desplegable de Heroku (com uns puntets) i ves a Data. Allà hauria d'apareixer la base de dades o que creis una (selecciona postgres, install postgres i et demanarà a quina app va relacionada). Un cop creada fes el migrate. Si tot va bé quan tornessis anar a Data t¡apareixeria el nom d'una base de dades i si cliques una mica el resum (espai que ocupa, nombre de taules que simplement comprova que no sigui 0). Si has de fer gaires vegades migrate sense exit o et peta la base de dades intenta fer un "Destroy" i tornar-la a crear. De vegades si hi ha molts errors la base de dades s'acaba fent un bunyol. 

Ja em diràs si has pogut solucionar el problema.

**Comment by jmcano94** [comment]
Ja tenia la base de dades de heroku creada, i quan intento fer el migrate, em dona exactament el mateix error. 


![screenshot_20180530_094807](https://user-images.githubusercontent.com/14343250/40706676-d92e3a0e-63ee-11e8-82c8-19527d2a58ce.png)
![screenshot_20180530_094951](https://user-images.githubusercontent.com/14343250/40706677-d94cbe7a-63ee-11e8-854d-c9deb98bd7b1.png)

**Comment by llevaNEUS** [comment]
Ostres... és tema que no troba la base de dades. Fes una cosa... carregat la base de dades. Elimina tot lo de pycache (*pyc) que puguis tenir en local i al github (millor si feu que no pugin a github). I fes el deployment sense la base de dades. Un cop la pàgina et funcioni en local i al heroku sense problemes. Comprova que segueixes el manual al peu de la lletra. Com per exemple que local_settings.py dins ykea i posat a .gitignore, que aquestes linies estiguin al final del document de settings.py:

try:
    from ykea.local_settings import *
except ImportError:
    # No local settings was found, skipping.
    pass
Després d'haver fet tot això, doncs creeu la base de dades i l'enllaceu i intenteu fer el deployment. 
Crec que deu ser que alguna cosa que ha quedat mal configurada.  

Aneu-me informant de com us va

**Comment by jmcano94** [comment]
Encara no funciona, he fet tot el que deies mes a dalt. Esborrat tots els arxius .pyc del repository, tant en local com en git. He esborrat la aplicació de heroku (la base de dades inclosa) y ho he tornat a pujar tot. Segueixo pensant que es problema de com li diem a heroku que ha de conectarse a la base de dades:
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql_psycopg2',
        'NAME': 'sd2018-ykea-b2',
        'USER': 'user',
        'PASSWORD': '',
        'HOST': 'localhost',
        'PORT': '',
        'CONN_MAX_AGE': 500
    }
}

**Comment by llevaNEUS** [comment]
Lo has de dir a kina app va associada i django-heroku fa la resta de la
magia es igual el nom que posis. Es molt estrany que no se't solucioni.

**Comment by jmcano94** [comment]
Jo no se que més probar ja.

**Comment by jmcano94** [comment]
Ja ho tinc, he buscat per internet una altre manera de fer-ho ho deixo aqui per a qui li passi el mateix y no ho aconsegueixi resoldre:

https://simpleisbetterthancomplex.com/tutorial/2016/08/09/how-to-deploy-django-applications-on-heroku.html

**Comment by jordipainan** [comment]
Mateix problema

**Comment by jmcano94** [comment]
@jordipainan Has mirat el tutorial que he passat mes a dalt?

**Comment by llevaNEUS** [comment]
El problema del @jordipainan era del fitxers de migrations i d'un possible tema d'incompatibilitat entre Django 1 i 2. Comproveu la versió que teniu instal.lada en local sigui < 2 i que per ser millor encara coincideixi amb la instal·lada a Heroku. Realment tampoc veig gran diferencia entre el tutorial que has passat i el que us hem dit que seguisiu... aix.. aquells grans misteris del Deployment! 
Bé, tot i així m'alegro que els problemes es vagin solucionant.

**Comment by jordipainan** [comment]
@jmcano94 Yes! Però no l'he fet servir, he tornat a fer el projecte de 0, eliminant la DB i tots els .pyc, tornar a fer els migrates y m'ha funcionat seguint el tutorial que hi ha a la pagina de l'assignatura.
